### PR TITLE
[Core] Add native ModelExpress load format

### PR DESCRIPTION
**Pull request by zhengluo-nv**
## Purpose

Add native vLLM registration for `--load-format modelexpress`.

This PR adds a thin in-tree `ModelExpressModelLoader` wrapper that delegates the actual ModelExpress loading implementation to the optional ModelExpress Python package. vLLM owns only the integration surface:

- registers `modelexpress` in the vLLM load-format registry
- exposes a clear missing-package error when ModelExpress is not installed
- delegates `download_model`, `load_weights`, and `load_model` to ModelExpress
- preserves the standard `Load model` tracing span and returns the loaded model in eval mode
- allows RunAI/object-storage `model_weights` validation when `load_format == "modelexpress"`

The ModelExpress package continues to own the real loader implementation, vLLM adapter behavior, and transfer strategies such as RDMA, ModelStreamer, GDS, and native fallback. The ModelExpress plugin path keeps `mx` as a backward-compatible alias outside vLLM. This PR includes AI-assisted code changes.

## Test Plan

- Verify `load_format="modelexpress"` resolves to the in-tree `ModelExpressModelLoader`.
- Verify the vLLM shim delegates `download_model`, `load_weights`, and `load_model` to the ModelExpress implementation.
- Verify the vLLM shim returns the delegated model in eval mode.
- Verify missing ModelExpress raises a clear install error.
- Verify unrelated import errors inside ModelExpress are preserved.
- Verify object-storage `model_weights` is accepted with `load_format="modelexpress"`.
- Verify end-to-end serving with both the native vLLM loader path and the older ModelExpress plugin path.
- Verify ModelExpress P2P transfer with a larger FP8 MoE model.

## Test Result

- Passed:
  - `PYTHONPYCACHEPREFIX=/private/tmp/vllm-pycache python3 -m py_compile vllm/model_executor/model_loader/modelexpress_loader.py tests/model_executor/model_loader/test_modelexpress_loader.py vllm/config/load.py vllm/config/vllm.py vllm/model_executor/model_loader/__init__.py`
- Passed locally before the reviewer-requested `mx` to `modelexpress` rename:
  - `uv run --no-project --with-requirements requirements/docs.txt mkdocs build --strict`
  - Result: `Documentation built in 354.64 seconds`
- End-to-end smoke test on nscale B200, run before the `modelexpress` rename against the same loader implementation:
  - Model: `Qwen/Qwen3.5-0.8B`
  - Native vLLM loader path: `vllm.model_executor.model_loader.mx_loader.MxModelLoader`
  - ModelExpress plugin path: `modelexpress.engines.vllm.loader.MxModelLoader`
  - Both paths accepted `--load-format mx`, loaded through the default/native fallback path when no MX server was configured, and completed a one-prompt generation job.
- End-to-end P2P test on nscale B200, run before the `modelexpress` rename against the same loader implementation:
  - Model: `mistralai/Mistral-Small-4-119B-2603`
  - Config: native vLLM loader, TP=4, DP=1, non-eager, `--language-model-only`, `--max-model-len 4096`, `--gpu-memory-utilization 0.50`
  - Source disk load: about 27s per worker, 1337 tensors / 30.49 GiB registered per TP rank, reached `Application startup complete`
  - Target P2P load: found one source worker per TP rank, transferred 30.49 GiB per rank in about 1.12-1.14s at about 214-218 Gbps, `MxModelLoader.load_model()` completed in about 10.5s
  - Target serving: `/v1/models` returned `mistralai/Mistral-Small-4-119B-2603`; `/v1/chat/completions` returned `four` for a simple arithmetic prompt
- Not run locally:
  - `uv run pytest tests/model_executor/model_loader/test_modelexpress_loader.py -q`
  - This macOS environment fails vLLM dependency resolution for `torch==2.11.0+cpu`.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm/model_executor/model_loader/modelexpress_loader.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        self,
+        vllm_config: VllmConfig,
+        model_config: ModelConfig,
+        prefix: str = "",
+    ) -> nn.Module:
+        return self._loader.load_model(
+            vllm_config=vllm_config,
+            model_config=model_config,
+            prefix=prefix,
+        )
```
[the remark this answers is not in the document]
Comment (zhengluo-nv) [reply vllm/model_executor/model_loader/modelexpress_loader.py:69]: Thanks, this is addressed in the latest push. The vLLM shim now wraps `load_model` with the standard `@instrument(span_name="Load model")` tracing span and returns `model.eval()` after delegating to the ModelExpress loader. I also updated the unit test to verify the delegated model is returned in eval mode.

File: vllm/config/vllm.py [hunk 2]
```diff
@@ -1900,12 +1900,13 @@ def try_verify_and_update_config(self):
                 )
                 self.load_config.load_format = "runai_streamer"
             elif self.load_config.load_format not in (
+                "mx",
```
Comment (zhengluo-nv) [inline vllm/config/vllm.py]: We need this because mx has model streamer integrated inside, so it can also support loading from object storage.

File: vllm/config/load.py [hunk 3]
```diff
@@ -55,6 +55,7 @@ class LoadConfig:
       https://github.com/ggml-org/ggml/blob/master/docs/gguf.md).
     - "mistral" will load weights from consolidated safetensors files used by
       Mistral models.
+    - "mx" will load weights using ModelExpress.
```
Comment (robertgshaw2-redhat) [inline vllm/config/load.py]: let's call it `modelexpress`. `mx` is confusing because it conflicts with `mxFP4 or MXFP8`, especially for `--load-format`

File: tests/model_executor/model_loader/test_modelexpress_loader.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (robertgshaw2-redhat) [inline tests/model_executor/model_loader/test_modelexpress_loader.py:1]: we probably should have an actual test that runs mx code right?

File: vllm/model_executor/model_loader/modelexpress_loader.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        "modelexpress",
+        "modelexpress.engines",
+        "modelexpress.engines.vllm",
+        _MODELEXPRESS_LOADER_MODULE,
+    }
+)
+
+
+def _missing_modelexpress_error() -> ImportError:
```
Comment (robertgshaw2-redhat) [inline vllm/model_executor/model_loader/modelexpress_loader.py:26]: should we just add this as a required package on cuda? why not?

Comment (zhengluo-nv) [reply vllm/model_executor/model_loader/modelexpress_loader.py:26]: We plan to release the final vLLM native integration in the next ModelExpress version, so the release version is not ready yet. We will add modelexpress as an optional package once it's released

File: vllm/config/load.py (hunk 3 above)
Comment (zhengluo-nv) [reply vllm/config/load.py]: Renamed in the new commit

File: tests/model_executor/model_loader/test_modelexpress_loader.py (hunk 4 above)
Comment (zhengluo-nv) [reply tests/model_executor/model_loader/test_modelexpress_loader.py:1]: We could add an integration test after the next release version of modelexpress is pushed to PyPI, which will include the proper native vLLM integration support. Will track this as an action item on our side.

**Review by robertgshaw2-redhat — APPROVED** [review]
